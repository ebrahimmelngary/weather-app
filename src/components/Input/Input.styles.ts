import {StyleSheet} from 'react-native';

import {COLORS} from '../../common/colors';

export default StyleSheet.create({
  textInputStyle: {
    backgroundColor: COLORS.white,
    padding: 10,
    borderRadius: 5,
  },
});
