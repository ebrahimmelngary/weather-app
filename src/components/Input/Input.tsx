import React from 'react';
import { TextInput, TextInputProps } from 'react-native';

import styles from './Input.styles';

const Input = (props: TextInputProps) => {
  return <TextInput testID='custom-input'{...props} style={styles.textInputStyle} />;
};

export default Input;
