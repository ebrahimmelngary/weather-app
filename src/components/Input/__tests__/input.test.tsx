import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Input from '../Input';

describe('Input', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<Input testID="custom-input" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByTestId } = render(<Input testID="custom-input" />);
    const inputComponent = getByTestId('custom-input');
    expect(inputComponent).toBeDefined();
  });

  it('renders with the provided style', () => {
    const { getByTestId } = render(<Input testID="custom-input" style={{ padding: 10 }} />);
    const inputComponent = getByTestId('custom-input');
    expect(inputComponent.props.style).toEqual(expect.objectContaining({ padding: 10 }));
  });

  it('handles onChange events', () => {
    const handleChange = jest.fn();
    const { getByTestId } = render(<Input testID="custom-input" onChangeText={handleChange} />);
    const inputComponent = getByTestId('custom-input');

    fireEvent.changeText(inputComponent, 'test input');
    expect(handleChange).toHaveBeenCalledWith('test input');
  });
});
