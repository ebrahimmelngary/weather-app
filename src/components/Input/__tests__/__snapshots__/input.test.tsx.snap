// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<TextInput
  style={
    {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "padding": 10,
    }
  }
  testID="custom-input"
/>
`;
